darshan-util/pydarshan/tests/test_report.py

```diff
@@ -63,6 +63,20 @@ def test_unsupported_record_load(caplog, unsupported_record):
         assert unsupported_record in record.message
 
 
+def test_invalid_dxt_mod():
+    """Invalid dxt module case"""
+
+    report = darshan.DarshanReport("tests/input/sample-dxt-simple.darshan")
+
+    report.mod_read_all_dxt_records("BUGGER")  # Intentional error.
+
+def test_unsupported_dxt_mod():
+    """Valid module but unsupported case"""
+
+    report = darshan.DarshanReport("tests/input/sample-dxt-simple.darshan")
+
+    report.mod_read_all_dxt_records("STDIO")
```

Comment:
A few comments:

- you haven't actually checked that the `logger` module issues the appropriate warning text (this is discussed in the matching issue)
- you could reduce some duplication and make failures easier to identify by using [parametrization](https://docs.pytest.org/en/6.2.x/parametrize.html)
- it looks like this only covers half of the code mentioned in the matching issue (checking locally on this branch with: `pytest --cov=darshan --cov-report=html`:
![image](https://user-images.githubusercontent.com/7903078/132740915-bc6895de-eb57-4d03-abc1-0bc6364f4ceb.png)

Codecov seems to concur as well:
![image](https://user-images.githubusercontent.com/7903078/132741008-e1d406fe-ac90-4e4d-a4ee-3099cd5724c5.png)